packages/react-components/react-table/src/hooks/selectionManager.ts

```diff
@@ -0,0 +1,112 @@
+import { SelectionMode } from './types';
+
+type OnSelectionChangeCallback = (selectedItems: Set<SelectionItemId>) => void;
+
+export interface SelectionManager {
+  toggleSelect(id: SelectionItemId): void;
+  select(id: SelectionItemId): void;
+  deSelect(id: SelectionItemId): void;
+  clearSelection(): void;
+  isSelected(id: SelectionItemId): boolean;
+  toggleSelectAll(itemIds: SelectionItemId[]): void;
+}
+
+export type SelectionItemId = string | number;
+
+export function createSelectionManager(
+  mode: SelectionMode,
+  onSelectionChange: OnSelectionChangeCallback = () => undefined,
+): SelectionManager {
+  const selectedItems = new Set<SelectionItemId>();
+  return mode === 'multiselect'
+    ? createMultipleSelectionManager(selectedItems, onSelectionChange)
+    : createSingleSelectionManager(selectedItems, onSelectionChange);
+}
+
+function createMultipleSelectionManager(
+  set: Set<SelectionItemId>,
```

Comment:
_nit_ Let's not use abstract "set"

```suggestion
  items: Set<SelectionItemId>,
```

or 

```suggestion
  itemsSet: Set<SelectionItemId>,
```

Reply:
went for `selectedItems`